### PR TITLE
 en, zh: refine docs: deploying TiFlash and heterogeneous clusters in Kubernetes

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added, or deleted? (Required)

Refine the following docs：
- deploy-tiflash.md
- deploy-heterogeneous-tidb-cluster.md

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 1]
```diff
@@ -5,112 +5,94 @@ summary: 本文档介绍如何为已有的 TiDB 集群部署一个异构集群
...
-
-```shell
-kubectl create -f tidbmonitor.yaml -n ${namespace}
-```
-
-## 部署 TLS 异构集群
+依据你是否需要为异构集群开启 TLS，请选择以下方案之一：
+
+- 部署未开启 TLS 的异构集群
+- 部署开启 TLS 的异构集群
```
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/deploy-heterogeneous-tidb-cluster.md:28]: 可以的，我试试。

File: zh/deploy-tiflash.md [hunk 2]
```diff
@@ -1,91 +1,159 @@
...
+title: 为已有 TiDB 集群部署 HTAP 存储引擎 TiFlash
+summary: 了解如何在 Kubernetes 上为已有 TiDB 集群部署 TiDB HTAP 存储引擎 TiFlash。
 aliases: ['/docs-cn/tidb-in-kubernetes/dev/deploy-tiflash/']
 ---
 
-# 在 Kubernetes 上部署 TiFlash
+# 为已有 TiDB 集群部署 HTAP 存储引擎 TiFlash
 
-本文介绍如何在 Kubernetes 上部署 TiFlash。
+本文介绍如何在已有的 TiDB 集群上部署或删除 TiDB HTAP 存储引擎 TiFlash。TiFlash 是 TiKV 的列存扩展，在提供了良好的隔离性的同时，也兼顾了与 TiKV 的强一致性。
```
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/deploy-tiflash.md:9]: 好的，那我把后面的相关链接挪过来。

**Review by csuzhangxc — APPROVED** [review]

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 3]
```diff
@@ -5,112 +5,101 @@ summary: 本文档介绍如何为已有的 TiDB 集群部署一个异构集群
...
 
-本文档介绍如何为已有的 TiDB 集群部署一个异构集群。
+本文介绍如何为已有的 TiDB 集群部署一个异构集群。异构集群是与已有 TiDB 集群不同配置的节点构成的集群。
+
+## 适用场景
+
+适用于基于已有的 TiDB 集群需要创建一个差异化配置的实例节点的场景，例如：
+
+- 创建不同配置不同 Label 的 TiKV 集群用于热点调度
+- 创建不同配置的 TiDB 集群分别用于 TP 和 AP 查询
```
Comment (ran-huang) [inline zh/deploy-heterogeneous-tidb-cluster.md:15]:
```suggestion
- 创建不同配置的 TiDB 集群分别用于 OLTP 和 OLAP 查询
```

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 4]
```diff
@@ -119,63 +108,112 @@ kubectl create -f tidbmonitor.yaml -n ${namespace}
...
+        baseImage: pingcap/tiflash
+        maxFailoverCount: 0
+        replicas: 1
+        storageClaims:
+          - resources:
+              requests:
+                storage: 100Gi
+    ```
+
+    其中，`spec.tlsCluster.enabled` 表示组件间是否开启 TLS，`spec.tidb.tlsClient.enabled` 表示 MySQL 客户端是否开启 TLS。详细的异构 TLS 集群配置示例，请参阅 ['heterogeneous-tls'](https://github.com/pingcap/tidb-operator/tree/master/examples/heterogeneous-tls)。
```
Comment (ran-huang) [inline zh/deploy-heterogeneous-tidb-cluster.md:164]:
```suggestion
    其中，`spec.tlsCluster.enabled` 表示组件间是否开启 TLS，`spec.tidb.tlsClient.enabled` 表示 MySQL 客户端是否开启 TLS。详细的异构 TLS 集群配置示例，请参阅 [`heterogeneous-tls`](https://github.com/pingcap/tidb-operator/tree/master/examples/heterogeneous-tls)。
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 5]
```diff
@@ -119,63 +108,112 @@ kubectl create -f tidbmonitor.yaml -n ${namespace}
...
+1. 编辑已有 TiDB 集群的 TidbMonitor Custom Resource (CR)：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl edit tm ${cluster_name} -n ${namespace}
+    ```
+
+2. 参考以下示例，替换 `${origin_cluster_name}` 为想要加入的集群名称，替换 `${heterogeneous_cluster_name}` 为异构集群名称：
+
```
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/deploy-heterogeneous-tidb-cluster.md:194]: 这里应该不需要，因为只需要用户基于现有 yaml 文件中，增加一个 clusters.name 参数写上异构集群名

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 6]
```diff
@@ -5,112 +5,101 @@ summary: 本文档介绍如何为已有的 TiDB 集群部署一个异构集群
...
-```
-
-## 部署 TLS 异构集群
+依据你是否需要为异构集群开启 TLS，请选择以下方案之一：
+
+- 部署未开启 TLS 的异构集群
+- 部署开启 TLS 的异构集群
+
+<SimpleTab>
+<div label="非 TLS">
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply zh/deploy-heterogeneous-tidb-cluster.md:31]: 默认 "非 TLS" 吧，我们部署时也是把 TLS 单独放在 "安全" 里去了

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 7]
```diff
@@ -5,112 +5,101 @@ summary: 本文档介绍如何为已有的 TiDB 集群部署一个异构集群
...
+    例如，你可以修改 `cluster.yaml` 文件中各组件的 `replicas` 数量或者删除不需要的组件。
+
+3. 执行以下命令创建异构集群。你需要将 `cluster.yaml` 替换为你的异构集群配置文件名。
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create -f cluster.yaml -n ${namespace}
+    ```
+
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply zh/deploy-heterogeneous-tidb-cluster.md:99]: `tidbcluster.pingcap.com/${heterogeneous_cluster_name} created`

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 8]
```diff
@@ -119,63 +108,112 @@ kubectl create -f tidbmonitor.yaml -n ${namespace}
...
+    例如，你可以修改 `cluster.yaml` 文件中各组件的 `replicas` 数量或者删除不需要的组件。
+
+3. 执行以下命令创建开启 TLS 的异构集群。你需要将 `cluster.yaml` 替换为你的异构集群配置文件名。
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    kubectl create -f cluster.yaml -n ${namespace}
+    ```
+
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply zh/deploy-heterogeneous-tidb-cluster.md:185]: `tidbcluster.pingcap.com/${heterogeneous_cluster_name} created`

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 9]
```diff
@@ -119,63 +108,112 @@ kubectl create -f tidbmonitor.yaml -n ${namespace}
...
+      timezone: UTC
+      pvReclaimPolicy: Delete
+      discovery: {}
+      cluster:
+        name: ${origin_cluster_name}
+      tikv:
+        baseImage: pingcap/tikv
+        maxFailoverCount: 0
+        replicas: 1
+        # if storageClassName is not set, the default Storage Class of the Kubernetes cluster will be used
```
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/deploy-heterogeneous-tidb-cluster.md]: 是的，已翻译。 PTAL

File: zh/deploy-heterogeneous-tidb-cluster.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/deploy-heterogeneous-tidb-cluster.md:185]: 已增加，PTAL

File: zh/deploy-heterogeneous-tidb-cluster.md [hunk 10]
```diff
@@ -119,63 +108,112 @@ kubectl create -f tidbmonitor.yaml -n ${namespace}
...
+创建证书后，要部署一个开启 TLS 的异构集群，请进行以下操作：
+
+1. 为异构集群新建一个集群配置文件。
+
+    例如，将如下配置存为 `cluster.yaml` 文件，并替换 `${heterogeneous_cluster_name}` 为自己想命名的异构集群名字，`${origin_cluster_name}` 替换为想要加入的已有集群名称。
+
+    > **注意**:
+    >
+    > 相比于普通 TiDB 集群配置文件，异构集群配置文件的唯一区别是，你需要额外配置 `spec.cluster.name` 字段为已有的 TiDB 集群名。通过此字段，TiDB Operator 会将该异构集群加入到已有的 TiDB 集群。
+
```
[the remark this answers is not in the document]
Comment (qiancai) [reply zh/deploy-heterogeneous-tidb-cluster.md:124]: 已增加，PTAL

File: en/deploy-heterogeneous-tidb-cluster.md [hunk 11]
```diff
@@ -1,114 +1,107 @@
...
 
-This document describes how to deploy a heterogeneous cluster for an existing TiDB cluster.
+This document describes how to deploy a heterogeneous cluster for an existing TiDB cluster. A heterogeneous cluster consists of nodes with different configurations from the existing TiDB cluster.
+
+## User scenarios
+
+This document is applicable to scenarios in which you need to create differentiated instances for an existing TiDB cluster, such as the following:
+
+- Create a TiKV cluster with different configurations and different labels for hotspot scheduling.
+- Create a TiDB cluster with different configurations for OLTP and OLAP queries.
```
Comment (hfxsd) [inline en/deploy-heterogeneous-tidb-cluster.md:15]:
```suggestion
- Create TiKV clusters with different configurations and different labels for hotspot scheduling.
- Create TiDB clusters with different configurations for OLTP and OLAP queries.
```
从原文来判断，似乎用复数 clusters 更合适？

File: en/deploy-tiflash.md [hunk 12]
```diff
@@ -1,93 +1,161 @@
...
+                log = "/data0/logs/flash_cluster_manager.log"
+            [logger]
+              count = 10
+              level = "information"
+              errorlog = "/data0/logs/error.log"
+              log = "/data0/logs/server.log"
+    ```
 
-For example:
+    For more TiFlash parameters that can be configured, refer to [TiFlash Configuration Documentation](https://pingcap.com/docs-cn/stable/tiflash/tiflash-configuration/).
```
[the remark this answers is not in the document]
Comment (qiancai) [reply en/deploy-tiflash.md:97]:
```suggestion
    For more TiFlash parameters that can be configured, refer to [TiFlash Configuration Documentation](https://docs.pingcap.com/tidb/stable/tiflash-configuration).
```

**Pull request merged**